Library/Homebrew/cask/dsl.rb

```diff
@@ -310,18 +310,36 @@ def version(arg = nil)
     # For architecture-dependent downloads:
     #
     # ```ruby
-    # sha256 arm:   "7bdb497080ffafdfd8cc94d8c62b004af1be9599e865e5555e456e2681e150ca",
-    #         intel: "b3c1c2442480a0219b9e05cf91d03385858c20f04b764ec08a3fa83d1b27e7b2"
+    # sha256 arm:          "7bdb497080ffafdfd8cc94d8c62b004af1be9599e865e5555e456e2681e150ca",
+    #        x86_64:       "b3c1c2442480a0219b9e05cf91d03385858c20f04b764ec08a3fa83d1b27e7b2"
+    #        x86_64_linux: "1a2aee7f1ddc999993d4d7d42a150c5e602bc17281678050b8ed79a0500cc90f"
+    #        arm64_linux:  "bd766af7e692afceb727a6f88e24e6e68d9882aeb3e8348412f6c03d96537c75"
     # ```
     #
     # @api public
-    def sha256(arg = nil, arm: nil, intel: nil)
-      should_return = arg.nil? && arm.nil? && intel.nil?
-
+    sig {
+      params(
+        arg:          T.nilable(T.any(String, Symbol)),
+        arm:          T.nilable(String),
+        intel:        T.nilable(String),
+        x86_64:       T.nilable(String),
+        x86_64_linux: T.nilable(String),
+        arm64_linux:  T.nilable(String),
+      ).returns(T.nilable(T.any(Symbol, Checksum)))
+    }
+    def sha256(arg = nil, arm: nil, intel: nil, x86_64: nil, x86_64_linux: nil, arm64_linux: nil)
+      should_return = arg.nil? && arm.nil? && (intel.nil? || x86_64.nil?) && x86_64_linux.nil? && arm64_linux.nil?
```

Comment:
```suggestion
      should_return = arg.nil? && arm.nil? && intel.nil? && x86_64.nil? && x86_64_linux.nil? && arm64_linux.nil?
```

Otherwise, I think the following case would return a value:

```ruby
sha256 intel: "<hash>"
```